src/VoyagerServiceProvider.php

```diff
@@ -88,6 +91,11 @@ public function boot(Router $router, Dispatcher $event)
             $this->addStorageSymlinkAlert();
         });
 
+        //Added a listener on event after update data
+        $event->listen('voyager.insert_update_data', function ($request, $slug, $rows, $data) {
```

Comment:
I'm also not sure I understand why you're using a trait on the ServiceProvider for this.  Is this a trait you're expecting to use elsewhere as well?